README.md
# adBottomArt

![](https://media.forgecdn.net/attachments/thumbnails/145/473/310/172/adBA.png)
![](https://media.forgecdn.net/attachments/thumbnails/145/477/310/172/adBA_GUI.png)

## Features
-   Displays an artwork at the bottom of the screen.
-   The first addon which auto load your skins, you don't need to change any codes.
-   You can quick switch your skin.
-   Auto Settings for 2048 or 1600 width skins.
-   Auto set skin height to same as chat height.
-   You can set the offset of your skin.
-   You can hide Chat and Triangle.
    -   Chat: All Chat background set opacity to 0 and Emotion, ChatReset Button set opacity to 0.1.
    -   Triangle: All the Triangle for move frame set opacity to 0.1.  
        ![Triangle](https://media.forgecdn.net/attachments/145/470/Triangle.png "Triangle")
-   You can hide MainMenu and Experience Bar.

## Add Your Skin
1.  Download skin from anywhere.
    -   The format of skin textures must fit adBottomArt.xml.
2.  Put your skin in ...\Interface\Addons\adBottomArt\Textures.
3.  Change the skin folder name to a continuous number. Ex: 1, 2, 3, 4, 5..., not 1, 3, 5....
4.  Copy ...\Interface\Addons\adBottomArt\Textures\1\loadfile.lua to your skin folder.
5.  Play game! You will find your skin in drop menu.

-   Your textures folder:

```　　- Interface  
　　　　- AddOns  
　　　　　　- adBottomArt  
　　　　　　　　- Textures  
　　　　　　　　　　+ snl  
　　　　　　　　　　+ 1  
　　　　　　　　　　+ 2  
　　　　　　　　　　 - 3  
　　　　　　　　　　　· bg1.tga  
　　　　　　　　　　　· bg2.tga  
　　　　　　　　　　　· bg3.tga  
　　　　　　　　　　　· bg4.tga  
　　　　　　　　　　　· loadfile.lua  
　　　　　　　　　　+ 4  
　　　　　　　　　　+ 5  
　　　　　　　　　　......
```
-   If you don't use above way, you can also set skin by /adba skin XXX.
-   There is must a folder which called snl in Textures folder in any version.
-   The max skins I limit at 30, if anyone want, I can change it to any number.

### You Can Download Skin from
-   [http://rom.curseforge.com/addons/magic-mirror-skin/](http://rom.curseforge.com/addons/magic-mirror-skin/)
-   [http://rom.curseforge.com/addons/adbottomart-wave-skin/](http://rom.curseforge.com/addons/adbottomart-wave-skin/)
-   [http://rom.curseforge.com/addons/woodsy-skin/](http://rom.curseforge.com/addons/woodsy-skin/)
-   [http://rom.curseforge.com/addons/meadow-skin-adbottomart/](http://rom.curseforge.com/addons/meadow-skin-adbottomart/)
-   [http://elthrai.de/index.php?option=com_phocadownload&view=category&id=1&Itemid=152](http://elthrai.de/index.php?option=com_phocadownload&view=category&id=1&Itemid=152)
-   [http://rom.curseforge.com/addons/vines/](http://rom.curseforge.com/addons/vines/)
-   [http://rom.curseforge.com/addons/marble-skin-adbottomart/](http://rom.curseforge.com/addons/marble-skin-adbottomart/)
-   [http://rom.curseforge.com/addons/angelic-skin-adbottomart/](http://rom.curseforge.com/addons/angelic-skin-adbottomart/)
-   [http://rom.curseforge.com/addons/heritage-skin-adbottomart/](http://rom.curseforge.com/addons/heritage-skin-adbottomart/)
-   [http://rom.curseforge.com/addons/demonic3-skin-adbottomart/](http://rom.curseforge.com/addons/demonic3-skin-adbottomart/)
-   [http://rom.curseforge.com/addons/minimalistic-skin-adbottomart/](http://rom.curseforge.com/addons/minimalistic-skin-adbottomart/)
-   [http://rom.curseforge.com/addons/xbar-iii/#c12](http://rom.curseforge.com/addons/xbar-iii/#c12)

## Slash Commands
-   **Press Enter to set or Tab to change editbox in GUI**.
-   /adBA or /adBottomArt - Open GUI.
-   /adBA alpha 0.5 - Change transparency of artwork to 0.5, varies between 0 ~ 1.0.
-   /adBA scale 2.2 - Change the scale of artwork to 2.2.
-   /adBA width 256 - Change the width of independent artworks to 256.
-   /adBA height 128 - Change the height of independent artworks to 128.
-   /adBA xoffset 10 - Change the x offset of artworks to 10.
-   /adBA yoffset 32 - Change the y offset of artworks to 32.
-   /adBA switch - Switch the bottom art.
-   /adBA skin XXX - Change skin which the folder called XXX.
-   /adBA autow - Set textures to fit your screen resolution.
-   /adBA autoh - Set the textures height to fit your screen resolution.
-   /adBA chath - Set the textures height to same as chat height.
-   /adBA 16center - Set the 1600 width textures to center.
-   /adBA 16scale - Set the 1600 width textures height and width to fit your screen resolution.
-   /adBA 16hath - Set the 1600 width textures height to same as chat height.

## Description
-   If you want to move Minimap, please use other addon.
-   The Information bar is not add by adBottomArt.
-   Sorry, I added 2 skins in the addon if you (the skin author) don't like, I will delete it.

## Do you have problem?
-   Please read [Addon Teaching](http://rom.curseforge.com/addons/adbottomart/pages/addon-teaching/) first.
-   Please report bug at [Tickets](http://rom.curseforge.com/addons/adbottomart/tickets/).




Changelog.txt
==========================================================================
adBottomArt v0.6
--------------------------------------------------------------------------
* Added setting for set 1600 skin to chat height.
* Added support for XBar Addon Management.
* Fixed text of drop menu.
* Changed loading time.
* Code enhanced.
* Changed color of loaded text and translated it.


==========================================================================
adBottomArt v0.51
--------------------------------------------------------------------------
* Fixed a bug when you use UI Scale in game.
* Auto Width will also set X offset to 0.


==========================================================================
adBottomArt v0.5
--------------------------------------------------------------------------
* Fixed the function to set offset.
* Changed min value in width.
* Added minus in front of Y offset Editbox.


==========================================================================
adBottomArt v0.4
--------------------------------------------------------------------------
* Added quick switch skin button.
* Added button to set skin height with chat height.
* Changed width, height, scale and transparency EditBox into Slider.
* Show width text as whole skin width.
* Get back BG5~8 skin support make it compatible with other skins.
* Deleted 1/4 screen height button.
* Simplify code.


==========================================================================
adBottomArt v0.32
--------------------------------------------------------------------------
* Found framelevel=-1, so adBA is the lowest level now.


==========================================================================
adBottomArt v0.31
--------------------------------------------------------------------------
* Change something for Ctrl+Z function in RoM.


==========================================================================
adBottomArt v0.3
--------------------------------------------------------------------------
* Added a function to hide "MainMenu" and "Experience Bar".
* Added a function to hide "Chat background" and "the Triangle for move frame".
* Added auto settings for skins with 1600 width.
* Fixed the bug when you use Tab to change Editbox focus.
* Fixed OnShow bug of drop menu.
* Merged 6 "Set" to one button.
* Changed config frame.


==========================================================================
adBottomArt v0.2
--------------------------------------------------------------------------
* Auto load your skin, you don't need to change any codes.
* Setted skin at center.
* Added X offset setting, X is start from center.
* Added Enable check button.
* Changed auto set for height.
* Added a button for set the art height to 1/4 screen height.
* Added two skins.


==========================================================================
adBottomArt v0.1
--------------------------------------------------------------------------
* Support translation.
* Added Y offset setting.
* Added skin set in Configuration.
* Added auto set for height.
* Deleted AutoScale.
* Update editbox texts when you use AutoWidth, slash command.
* Press enter to set, not ClearFocus.
* Tab or Shift + Tab to change editbox.
* Fixed the editbox width, it will not cover Set button.
* Added a tooltip on editbox.
* Changed some offset in Configuration.


==========================================================================
Sorry, I don't want use Subversion to update.
==========================================================================
------------------------------------------------------------------------
r3 | anahel | 2009-04-09 11:42:09 +0000 (Thu, 09 Apr 2009) | 1 line
Changed paths:
   M /trunk/adBottomArt.lua
   M /trunk/adBottomArt.toc
   A /trunk/adBottomArtConfig.xml

added a config window
------------------------------------------------------------------------
r2 | anahel | 2009-04-08 22:08:05 +0000 (Wed, 08 Apr 2009) | 2 lines
Changed paths:
   A /trunk/Textures
   A /trunk/Textures/snl
   A /trunk/Textures/snl/bg1.tga
   A /trunk/Textures/snl/bg2.tga
   A /trunk/Textures/snl/bg3.tga
   A /trunk/Textures/snl/bg4.tga
   A /trunk/adBottomArt.lua
   A /trunk/adBottomArt.toc
   A /trunk/adBottomArt.xml

Initial release as repository
Added autoscale function
------------------------------------------------------------------------
r1 | root | 2009-04-08 22:03:29 +0000 (Wed, 08 Apr 2009) | 1 line
Changed paths:
   A /branches
   A /tags
   A /trunk

"adbottomart/mainline: Initial Import"
------------------------------------------------------------------------


Locales/DE.lua

adBALang = {
	["HideDesc"]="Hide Frames......",
	["ToggleMenu"]="MainMenu",
	["ToggleChat"]="Chat & Triangle",
	["Skin"]="Hintergrund",
	["SwitchSkin"]="Quick Switch Skin",
	["AddDesc"]="The best scale when you use auto setting is 1.0.",
	["AutoWidth"]="Auto Breite",
	["AutoHeight"]="Auto Höhe",
	["ChatHeight"]="Chat Höhe",
	["1600Desc"]="For skin width 1600",
	["1600Center"]="To Center",
	["1600CenterDesc"]="After you clicked AutoBreite, and don't want to change width again, you can use the button to set the skin to center.",
	["1600Scale"]="Auto Maßstab",
	["1600ScaleDesc"]="Auto set width/height to fit screen.\n(It will change X offset.)",
	["XAddonTip1"]="Left-Click to open GUI.",
	["XAddonTip2"]="Right-Click to toggle ExtraTip.",
	["Load"]="|cffEEABABadBottomArt %s|r geladen, you have |cffEEABAB%s|r skins, |cffEEABAB/adba|r eingeben um Einstellungen zu öffnen.",
};

Locales/ES.lua
--translated by Tschuss
adBALang = {
	["HideDesc"]="Hide Frames......",
	["ToggleMenu"]="MainMenu",
	["ToggleChat"]="Chat & Triangle",
	["Skin"]="fondo",
	["SwitchSkin"]="Quick Switch Skin",
	["AddDesc"]="The best scale when you use auto setting is 1.0.",
	["AutoWidth"]="Auto-ancho",
	["AutoHeight"]="Auto-Alto",
	["ChatHeight"]="Chat Height",
	["1600Desc"]="For skin width 1600",
	["1600Center"]="To Center",
	["1600CenterDesc"]="After you clicked Auto-ancho, and don't want to change width again, you can use the button to set the skin to center.",
	["1600Scale"]="Auto-escala",
	["1600ScaleDesc"]="Auto set width/height to fit screen.\n(It will change X offset.)",
	["XAddonTip1"]="Left-Click to open GUI.",
	["XAddonTip2"]="Right-Click to toggle adBottomArt.",
	["Load"]="|cffEEABABadBottomArt %s|r loaded, you have |cffEEABAB%s|r skins, type |cffEEABAB/adba|r to config.",
};

Locales/FR.lua
--translated by Gonzo1968
adBALang = {
	["HideDesc"]="Hide Frames......",
	["ToggleMenu"]="MainMenu",
	["ToggleChat"]="Chat & Triangle",
	["Skin"]="arri\195\168re-plan",
	["SwitchSkin"]="Quick Switch Skin",
	["AddDesc"]="The best scale when you use auto setting is 1.0.",
	["AutoWidth"]="LargeurAuto",
	["AutoHeight"]="HauteurAuto",
	["ChatHeight"]="Chat Height",
	["1600Desc"]="For skin width 1600",
	["1600Center"]="To Center",
	["1600CenterDesc"]="After you clicked LargeurAuto, and don't want to change width again, you can use the button to set the skin to center.",
	["1600Scale"]="\195\136chelleAuto",
	["1600ScaleDesc"]="Auto set width/height to fit screen.\n(It will change X offset.)",
	["XAddonTip1"]="Left-Click to open GUI.",
	["XAddonTip2"]="Right-Click to toggle adBottomArt.",
	["Load"]="|cffEEABABadBottomArt %s|r loaded, you have |cffEEABAB%s|r skins, type |cffEEABAB/adba|r to config.",
};

Locales/ENUS.lua

adBALang = {
	["HideDesc"]="Hide Frames......",
	["ToggleMenu"]="MainMenu",
	["ToggleChat"]="Chat & Triangle",
	["Skin"]="Skin",
	["SwitchSkin"]="Quick Switch Skin",
	["AddDesc"]="The best scale when you use auto setting is 1.0.",
	["AutoWidth"]="Auto Width",
	["AutoHeight"]="Auto Height",
	["ChatHeight"]="Chat Height",
	["1600Desc"]="For skin width 1600",
	["1600Center"]="To Center",
	["1600CenterDesc"]="After you clicked AutoWidth, and don't want to change width again, you can use the button to set the skin to center.",
	["1600Scale"]="Auto Scale",
	["1600ScaleDesc"]="Auto set width/height to fit screen.\n(It will change X offset.)",
	["XAddonTip1"]="Left-Click to open GUI.",
	["XAddonTip2"]="Right-Click to toggle adBottomArt.",
	["XAddonTip2"]="Right-Click to toggle adBottomArt.",
	["Load"]="|cffEEABABadBottomArt %s|r loaded, you have |cffEEABAB%s|r skins, type |cffEEABAB/adba|r to config.",
};

adBottomArt.lua
_G.ADBAVERSION = 0.63;
adBASet = {};
local maxtex = 30;
local add={
	gui={{
		name = "adBottomArt",
		version = ADBAVERSION,
		window = "adBottomArtGUI",
	}},
	popup={{
		GetText = function() return adBottomArt_PopText(); end,
		GetTooltip = function() return "/adba, /adBottomArt\n\n"..adBALang["XAddonTip1"].."\n"..adBALang["XAddonTip2"]; end,
		OnClick = function(this, key) adBottomArt_PopClick(this, key); end,
}}};

function adBottomArt_PopClick(this, key)
	if key=="LBUTTON" then
		XAddon_ShowPage("adBottomArtGUI");
	else
		adBottomArtSlash(nil, "switch");
	end
end

function adBottomArt_PopText()
	if adBASet["ArtShown"]==true then
		return "adBottomArt - "..ON;
	else
		return "adBottomArt - "..C_OFF;
	end
end

function adBottomArt_OnLoad(this)
	this:RegisterEvent("VARIABLES_LOADED");
	adBottomArtDownMenu_OnLoad();

	local lang = GetLanguage():upper();
	local _, err = loadfile("Interface/Addons/adBottomArt/Locales/"..lang..".lua");
	if err then
		adBottomArt_Print("|cff993333adBA can't find translation, ENUS.lua loaded.|r");
		dofile("Interface/Addons/adBottomArt/Locales/ENUS.lua");
	else
		dofile("Interface/Addons/adBottomArt/Locales/"..lang..".lua");
	end
end

function adBottomArtDownMenu_OnLoad()
	for i = 1, maxtex do
		local _, err = loadfile("Interface/Addons/adBottomArt/Textures/"..i.."/loadfile.lua");
		if err then
			maxtex = i - 1;
			return;
		end
	end
end

function adBottomArtDownMenu_OnShow()
	local info = {};
	info.value = 0;
	info.text = "snl";
	info.func = adBottomArtDownMenu_Click;
	UIDropDownMenu_AddButton(info);

	for i = 1, maxtex do
		info = {};
		info.value = i;
		info.text = i;
		info.func = adBottomArtDownMenu_Click;
		UIDropDownMenu_AddButton(info);
	end
end

function adBottomArtDownMenu_Click(button)
	if button.value==0 then
		adBASet["ArtName"] = "snl";
	else
		adBASet["ArtName"] = button.value;
	end
	adBottomArtSetSkin(adBASet["ArtName"]);
	UIDropDownMenu_SetSelectedID(adBottomArtGUIDownMenu, button.value+1);
end

function adBottomArtSwitch_Click()
	if adBASet["ArtName"]==maxtex then
		adBottomArtSetSkin("snl");
		adBASet["ArtName"] = "snl";
		UIDropDownMenu_SetSelectedID(adBottomArtGUIDownMenu, 1);
	return; end
	local skin = 1;
	for i = 1, maxtex do
		if adBASet["ArtName"]==i then
			skin = i+1;
		end
	end
	adBottomArtSetSkin(skin);
	adBASet["ArtName"] = tonumber(skin);
	UIDropDownMenu_SetSelectedID(adBottomArtGUIDownMenu, skin+1);
end

function adBottomArtSetWidth(v)
	local scale=GetUIScale();
	for i = 1, 8 do
		getglobal("adBottomArtFrameBG"..i):SetWidth(v/scale);
	end
end

function adBottomArtSetHeight(v)
	local scale=GetUIScale();
	for i = 1, 8 do
		getglobal("adBottomArtFrameBG"..i):SetHeight(v/scale);
	end
end

function adBottomArtSetSkin(v)
	for i = 1, 8 do
		getglobal("adBottomArtFrameBG"..i):SetTexture("Interface/AddOns/adBottomArt/Textures/"..v.."/bg"..i..".tga");
	end
end

function adBottomArtSetColor(v)
end

function adBottomArt_LoadSettings()
	local def = {
		["ToggleMenu"] = false,
		["ToggleChat"] = true,
		["ArtScale"] = 1,
		["ArtAlpha"] = 1,
		["ArtWidth"] = 512,
		["ArtHeight"] = 256,
		["ArtXOffset"] = 0,
		["ArtYOffset"] = 0,
		["ArtName"] = "snl",
		["ArtShown"] = true,
	};
	for k,v in pairs(def) do
		if adBASet[k]==nil then
			adBASet[k] = v;
		end
	end
	SaveVariables("adBASet");
end

function adBottomArt_OnEvent(event, this)
	if event=="VARIABLES_LOADED" then
		adBottomArt_LoadSettings();
		adBottomArt_Print(adBALang["Load"], ADBAVERSION, (maxtex+1));
		if adBASet["ToggleMenu"] ~= nil and adBASet["ToggleMenu"]==true then MainMenuFrame:Hide(); ExperienceFrame:Hide(); end
		if adBASet["ToggleChat"] ~= nil and adBASet["ToggleChat"]==true then
			for i = 1, 8 do
				getglobal("ChatFrame"..i):SetBackdropTileAlpha(0);
				getglobal("ChatFrame"..i):SetBackdropEdgeAlpha(0);
				getglobal("ChatFrame"..i.."ResizeButton"):SetAlpha(0.1);
			end
			ChatFrameEmoteButton:SetAlpha(0.1); ChatFrameRestoreButton:SetAlpha(0.1); ChatFrameChannelChangeResizeButton:SetAlpha(0.1);
			MainActionBarFrameButtonEX:SetAlpha(0.1); BottomActionBarFrameButtonEX:SetAlpha(0.1); LeftActionBarFrameButtonEX:SetAlpha(0.1); RightActionBarFrameButtonEX:SetAlpha(0.1);
		end
		if adBASet["ArtScale"] ~= nil then adBottomArtFrame:SetScale(adBASet["ArtScale"]); end
		if adBASet["ArtAlpha"] ~= nil then adBottomArtFrame:SetAlpha(adBASet["ArtAlpha"]); end
		if adBASet["ArtWidth"] ~= nil then adBottomArtSetWidth(adBASet["ArtWidth"]); end
		if adBASet["ArtHeight"] ~= nil then adBottomArtSetHeight(adBASet["ArtHeight"]); end
		if adBASet["ArtYOffset"] ~= nil and adBASet["ArtXOffset"] ~= nil then
			adBottomArtFrame:ClearAllAnchors();
			adBottomArtFrame:SetAnchor("BOTTOM", "BOTTOM", "UIParent", adBASet["ArtXOffset"]/adBASet["ArtScale"], -(adBASet["ArtYOffset"]/adBASet["ArtScale"]));
		end
		if adBASet["ArtName"] ~= nil then adBottomArtSetSkin(adBASet["ArtName"]); end
		if adBASet["ArtShown"] ~= nil and adBASet["ArtShown"]==false then adBottomArtFrame:Hide(); end
		if XBARVERSION and XBARVERSION>=1.51 then
			XAddon_Register(add);
		end
	end
end

function adBottomArt_GetWords(str)
	local ret = {};
	local pos=0;
	while(true) do
		local word;
		_,pos,word=string.find(str, "^ *([^%s]+) *", pos+1);
		if(not word) then
			return ret;
		end
		table.insert(ret, word);
	end
end

function adBottomArtSlash(ebox, v)
	local v = string.lower(v);
	local wL = adBottomArt_GetWords(v);
	local scale=GetUIScale();

	if wL[1]=="alpha" then
		adBASet["ArtAlpha"] = wL[2];
		adBottomArtFrame:SetAlpha(wL[2]);
	elseif wL[1]=="scale" then
		adBASet["ArtScale"] = wL[2];
		adBottomArtFrame:SetScale(wL[2]);
	elseif wL[1]=="width" then
		adBASet["ArtWidth"] = wL[2];
		adBottomArtSetWidth(wL[2]);
	elseif wL[1]=="height" then
		adBASet["ArtHeight"] = wL[2];
		adBottomArtSetHeight(wL[2]);
	elseif wL[1]=="xoffset" then
		adBASet["ArtXOffset"] = wL[2];
		adBottomArtFrame:ClearAllAnchors();
		adBottomArtFrame:SetAnchor("BOTTOM", "BOTTOM", "UIParent", adBASet["ArtXOffset"]/adBASet["ArtScale"], -(adBASet["ArtYOffset"]/adBASet["ArtScale"]));
	elseif wL[1]=="yoffset" then
		adBASet["ArtYOffset"] = wL[2];
		adBottomArtFrame:ClearAllAnchors();
		adBottomArtFrame:SetAnchor("BOTTOM", "BOTTOM", "UIParent", adBASet["ArtXOffset"]/adBASet["ArtScale"], -(adBASet["ArtYOffset"]/adBASet["ArtScale"]));
	elseif wL[1]=="switch" then
		adBASet["ArtShown"] = adBottomArtGUIEnable:IsChecked();
		if adBottomArtFrame:IsVisible() then
			adBASet["ArtShown"] = false;
			adBottomArtFrame:Hide();
		else
			adBASet["ArtShown"] = true;
			adBottomArtFrame:Show();
		end
	elseif wL[1]=="skin" then
		adBASet["ArtName"] = wL[2];
		adBottomArtSetSkin(wL[2]);
	elseif wL[1]=="autow" then
		adBottomArtSlash(nil, "width " .. tostring(512 * GetScreenWidth() / 2048)); --512 * scale
		adBottomArtSlash(nil, "scale 1");
		adBottomArtSlash(nil, "xoffset "..tostring(0) ); --1600 scale
	elseif wL[1]=="autoh" then
		adBottomArtSlash(nil, "height "..tostring(GetScreenWidth()/8) ); --4 parts of skin, width:screen = 1:4, height:width = 1:2
		adBottomArtSlash(nil, "scale 1");
	elseif wL[1]=="chath" then
		adBottomArtSlash(nil, "height "..tostring(ChatFrame1:GetHeight()+60 )*scale  ); --looks better
		adBottomArtSlash(nil, "scale 1");
	elseif wL[1]=="16center" then
		adBottomArtSlash(nil, "xoffset "..tostring( (224*(GetScreenWidth()/2048))/scale )  ); --1024-800=224, 1024->2048center, 800->1600center
		adBottomArtSlash(nil, "scale 1");
	elseif wL[1]=="16scale" then
		adBottomArtSlash(nil, "width "..tostring( (GetScreenWidth()/4)*1.28) ); --2048/1600 = 1.28
		adBottomArtSlash(nil, "height "..tostring( (GetScreenWidth()/8)*1.28) );
		adBottomArtSlash(nil, "xoffset "..tostring( (224*(GetScreenWidth()/1600))/scale )  ); --1600 scale
		adBottomArtFrame:SetScale(1);
	elseif wL[1]=="16chath" then
		adBottomArtSlash(nil, "height "..tostring(ChatFrame1:GetHeight()+60 )*scale*1.28  ); --looks better
		adBottomArtSlash(nil, "scale 1");
	else
		if XBARVERSION and XBARVERSION>=1.51 then
			XAddon_ShowPage("adBottomArtGUI");
		else
			ToggleUIFrame(adBottomArtGUI);
		end
	end
	adBottomArtTexts(adBottomArtGUI);
end

SLASH_adBottomArtSlash1 = "/adBottomArt";
SLASH_adBottomArtSlash2 = "/adBA";
SlashCmdList["adBottomArtSlash"] = adBottomArtSlash;

function adBottomArtTexts(this)
	adBottomArtGUIEnable:SetChecked(adBASet["ArtShown"]);
	adBottomArtGUIToggleMenu:SetChecked(adBASet["ToggleMenu"]);
	adBottomArtGUIToggleChat:SetChecked(adBASet["ToggleChat"]);
	adBottomArtGUIWidth:SetValue(adBASet["ArtWidth"]);
	adBottomArtGUIHeight:SetValue(adBASet["ArtHeight"]);
	adBottomArtGUIScale:SetValue(adBASet["ArtScale"]);
	adBottomArtGUIAlpha:SetValue(adBASet["ArtAlpha"]);
	adBottomArtGUIXOffset:SetText(adBASet["ArtXOffset"]);
	adBottomArtGUIYOffset:SetText(adBASet["ArtYOffset"]);

	adBottomArtGUITitle:SetText("adBottomArt")
	adBottomArtGUIHideDesc:SetText(adBALang["HideDesc"]);
	adBottomArtGUIToggleMenuName:SetText(adBALang["ToggleMenu"]);
	adBottomArtGUIToggleChatName:SetText(adBALang["ToggleChat"]);
	adBottomArtGUIAddDesc:SetText(adBALang["AddDesc"]);
	adBottomArtGUIXOffsetName:SetText("X");
	adBottomArtGUIYOffsetName:SetText("Y");
	adBottomArtGUIDownMenuName:SetText(adBALang["Skin"]);
	adBottomArtGUIAutoWidthName:SetText(adBALang["AutoWidth"]);
	adBottomArtGUIAutoHeightName:SetText(adBALang["AutoHeight"]);
	adBottomArtGUIChatHeightName:SetText(adBALang["ChatHeight"]);
	adBottomArtGUI1600Desc:SetText(adBALang["1600Desc"]);
	adBottomArtGUI1600CenterName:SetText(adBALang["1600Center"]);
	adBottomArtGUI1600ScaleName:SetText(adBALang["1600Scale"]);
	adBottomArtGUI1600ChatHeightName:SetText(adBALang["ChatHeight"]);
	adBottomArtGUIVersion:SetText(ADBAVERSION);
	if XBARVERSION and XBARVERSION>=1.51 then
		XAddon_Page(this);
		XAddon_HideBack1(this);
		adBottomArtGUITitle:SetText(" ");
		adBottomArtGUIVersion:SetText(" ");
	end
end

function adBottomArt_Print(str, ...)
	DEFAULT_CHAT_FRAME:AddMessage(str:format(...), 1, 1, 1);
end